realsense/hand/win/hand_module_object.cc

```diff
@@ -0,0 +1,611 @@
+    return;
+  }
+
+  PXCHandData::IHand* pxc_hand = NULL;
+  if (PXC_FAILED(pxc_hand_data_->QueryHandDataById(
+      params->hand_id, pxc_hand))) {
+    info->PostResult(CreateDOMException("Failed to get hand data by id.",
+                                        ERROR_NAME_ABORTERROR));
+    return;
+  }
+
+  PXCImage* image;
+  if (PXC_FAILED(pxc_hand->QuerySegmentationImage(image))) {
+    info->PostResult(CreateDOMException("Failed to get segmented image.",
+                                        ERROR_NAME_ABORTERROR));
+    return;
+  }
+
+  if (!MakeBinaryMessageForImage<uint8>(image)) {
+    info->PostResult(CreateDOMException("Failed to copy image data.",
+                                        ERROR_NAME_ABORTERROR));
+    return;
+  }
+
+  scoped_ptr<base::ListValue> result(new base::ListValue());
+  result->Append(base::BinaryValue::CreateWithCopiedBuffer(
+      reinterpret_cast<const char*>(binary_message_.get()),
+      binary_message_size_));
+  info->PostResult(result.Pass());
+}
+
+void HandModuleObject::OnGetContoursById(
+      scoped_ptr<XWalkExtensionFunctionInfo> info) {
+  scoped_ptr<GetContoursById::Params> params(
+      GetContoursById::Params::Create(*info->arguments()));
+  if (!params) {
+    info->PostResult(CreateDOMException("The parameter is not supported.",
+                                        ERROR_NAME_INVALIDACCESSERROR));
+    return;
+  }
+
+  if (!pxc_hand_data_) {
+    info->PostResult(CreateDOMException("No hand data.",
+                                        ERROR_NAME_NOTFOUNDERROR));
+    return;
+  }
+
+  PXCHandData::IHand* pxc_hand = NULL;
+  if (PXC_FAILED(pxc_hand_data_->QueryHandDataById(
+      params->hand_id, pxc_hand))) {
+    info->PostResult(CreateDOMException("Cannot get hand data by id.",
+                                        ERROR_NAME_ABORTERROR));
+    return;
+  }
+
+  std::vector<linked_ptr<Contour> > contours;
+
+  int number_of_contours = pxc_hand->QueryNumberOfContours();
+  for (int i = 0; i < number_of_contours; ++i) {
+    PXCHandData::IContour* pxc_contour;
+    if (PXC_FAILED(pxc_hand->QueryContour(i, pxc_contour)))
+      continue;
+    linked_ptr<Contour> js_contour(new Contour);
+    js_contour->is_outter = pxc_contour->IsOuter() ? true : false;
+    int points_size = pxc_contour->QuerySize();
+    if (points_size > 0) {
+      scoped_ptr<PXCPointI32[]> points(new PXCPointI32[points_size]);
+      if (PXC_FAILED(pxc_contour->QueryPoints(points_size, points.get())))
+        continue;
+      for (int j = 0; j < points_size; ++j) {
+        linked_ptr<Point2D> js_point(new Point2D);
+        js_point->x = points[j].x;
+        js_point->y = points[j].y;
+        js_contour->points.push_back(js_point);
+      }
+    }
+    contours.push_back(js_contour);
+  }
+
+  info->PostResult(GetContoursById::Results::Create(contours));
+}
+
+template <typename T>
+bool HandModuleObject::MakeBinaryMessageForImage(PXCImage* image) {
```

Comment:
I wish this could be extracted to common utils, as this is a common process for converting PXCImage to binary message